#
# patch deals with test failures caused by two factors:
#	- poor coding style in 3rd party ltmain.sh,
#	  which uses variables, without defining them first
#
#	- paranoid setting of our gmake test target, which
#	  runs sub-shells with 'set -u' option, which specifically
#	  asks to define variable before its first use
#
# there are two ways to fix it:
#	1) relax our build environment for test target
#
#	2) fix ltmain.sh which comes from upstream
#
# patch below opts for 2)
#
--- libtasn1-4.21.0/build-aux/ltmain.sh
+++ libtasn1-4.21.0/build-aux/ltmain.sh
@@ -5505,6 +5505,8 @@ DUALCASE=1; export DUALCASE # for MKS sh
 relink_command=\"$relink_command\"
 
 # This environment variable determines our operation mode.
+libtool_install_magic=\${libtool_install_magic:-'0'};
+libtool_execute_magic=\${libtool_execute_magic:-'0'};
 if test \"\$libtool_install_magic\" = \"$magic\"; then
   # install mode needs the following variables:
   generated_by_libtool_version='$macro_version'
@@ -5736,6 +5738,7 @@ func_exec_program ()
 	if test yes = "$shlibpath_overrides_runpath" && test -n "$shlibpath_var" && test -n "$temp_rpath"; then
 	  $ECHO "\
     # Add our own library path to $shlibpath_var
+    $shlibpath_var=\${$shlibpath_var:-''}
     $shlibpath_var=\"$temp_rpath\$$shlibpath_var\"
 
     # Some systems cannot cope with colon-terminated $shlibpath_var
